tests/suites/test_suite_psa_crypto_slot_management.function

```diff
@@ -90,8 +90,10 @@ static int invalidate_psa(invalidate_method_t invalidate_method)
             break;
     }
 
-    PSA_ASSERT(psa_crypto_init());
+    /* When AES_C is not defined CTR_DRBG relies on PSA to get AES-ECB so it
+     * holds an open key once psa_crypto_init() is called. */
     ASSERT_PSA_PRISTINE();
+    PSA_ASSERT(psa_crypto_init());
```

Comment:
This change bothers me. It should always be the case that this succeeds:
```
// Assume PSA not yet initialized
PSA_ASSERT(psa_crypto_init());
ASSERT_PSA_PRISTINE();
```
There are currently several places that test this: here, and in `test_suite_psa_crypto_init` by calling `PSA_DONE` to assert that PSA is pristine. That last category has been fixed by making `PSA_DONE` shut down the RNG before checking the slot count, which bothers me for other reasons.

I think the proper fix would be to make the key used by CTR_DRBG not count for `ASSERT_PSA_PRISTINE`, i.e. it should not count for `mbedtls_test_helper_is_psa_leaking`. We should change the logic in that function to allow one key slot if the DRBG is using a key slot.